### PR TITLE
dialects: (arith) add truncation option to ConstantOp

### PR DESCRIPTION
**Pull request by alexarice**
split off from #3583

### CODE REVIEW

File: uv.lock
Comment (superlopuh) [inline uv.lock:1]: uh oh, let's separate this out?

Comment (superlopuh) [reply uv.lock:1]: This is my bad, I think the new workflow should be to make this change a part of the dependabot PR, I'll do this quickly

File: xdsl/dialects/arith.py [hunk 1]
```diff
@@ -166,13 +166,18 @@ def __init__(
...
+        truncate_bits: bool = False,
     ) -> ConstantOp:
         if isinstance(value_type, int):
             value_type = IntegerType(value_type)
         return ConstantOp.create(
             result_types=[value_type],
-            properties={"value": IntegerAttr(value, value_type)},
+            properties={
+                "value": IntegerAttr(value, value_type, truncate_bits=truncate_bits)
+            },
```
Comment (superlopuh) [inline xdsl/dialects/arith.py:180]: could you please add a unit test for this?

File: uv.lock
Comment (superlopuh) [reply uv.lock:1]: Ah no I merged textual, you have a newer version of ruff than our requirements

Comment (alexarice) [reply uv.lock:1]: This is not ideal that switching between branches messes up lock files

**Review by superlopuh — APPROVED** [review]
TY

File: uv.lock
Comment (superlopuh) [reply uv.lock:1]: In theory, this is exactly what UV is supposed to enable, switching branches with different dependencies and updating those silently

**Pull request merged**